### PR TITLE
Improve fragment rendering logic to include JS and CSS resources

### LINKED ISSUE #129: JS, CSS resources included in a fragment are not added when the fragment is directly rendered
Consider following fragment:
```hbs
{{css "css/some-styles.css"}}
{{js "js/some-script.js"}}
<div>
     <p> some HTML content</p>
</div>
```
If this fragment rendered directly from client-side (via ajax), CSS and JS resources won't be added to the output HTML.

### PR DESCRIPTION
**Pull request by cnapagoda**
With this fix JS and CSS resources are included when a fragment is directly rendered

RESOLVE #129

### CODE REVIEW

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java [hunk 1]
```diff
@@ -217,7 +219,16 @@ public String renderFragment(HttpRequest request, HttpResponse response) {
         Model model = new MapModel(request.getFormParams());
         RequestLookup requestLookup = createRequestLookup(request, response);
         API api = new API(sessionRegistry, requestLookup);
-        return fragment.render(model, lookup, requestLookup, api);
+
+        JsonObject renderedFragment = new JsonObject();
+        renderedFragment.addProperty("html", fragment.render(model, lookup, requestLookup, api));
+        renderedFragment.addProperty(Placeholder.headJs.name(),
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: We are returning a JSON here.

Comment (manuranga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: @this not clear what you mean

Comment (manuranga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: Ok now I get it, My bad.

Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: We are returning a JSON here (`JsonObject`). IMO, creating a new class (with setJs, setCss etc.) just for this method is not a good way

**Comment by this** [comment]
Closing this PR due to commits complications. Changes that this PR should carry are available in PR #178.

**Pull request closed without merging**